src/Stratis.Bitcoin.IntegrationTests/Miners/ProofOfStakeMintCoinsSteps.cs

```diff
@@ -1,170 +1,73 @@
-                FeeType.Medium,
-                101);
-
-            context.OverrideFeeRate = new FeeRate(Money.Satoshis(20000));
-
-            var unspent = this.nodes[PowMiner].FullNode.WalletManager().GetSpendableTransactionsInWallet(PowWallet);
-            var coins = new List<Coin>();
-
-            var blockTimestamp = unspent.OrderBy(u => u.Transaction.CreationTime).Select(ts => ts.Transaction.CreationTime).First();
-            var transaction = new Transaction
-            {
-                Time = (uint)blockTimestamp.ToUnixTimeSeconds()
-            };
-
-            foreach (var item in unspent.OrderByDescending(a => a.Transaction.Amount))
-            {
-                coins.Add(new Coin(item.Transaction.Id, (uint)item.Transaction.Index, item.Transaction.Amount, item.Transaction.ScriptPubKey));
-            }
-
-            var coin = coins.First();
-            var txIn = transaction.AddInput(new TxIn(coin.Outpoint, this.powSenderAddress.ScriptPubKey));
-            transaction.AddOutput(new TxOut(new Money(9699999999995400), this.powSenderAddress.ScriptPubKey));
-            transaction.AddOutput(new TxOut(new Money(100000000000000), this.posReceiverAddress.ScriptPubKey));
-            transaction.Sign(this.powSenderPrivateKey, new[] { coin });
-
-            this.nodes[PowMiner].FullNode.NodeService<WalletController>().SendTransaction(new SendTransactionRequest(transaction.ToHex()));
+            this.proofOfStakeSteps.SendMillionCoinsFromPowWalletToPosWallet();
         }
 
         private void pow_wallet_broadcasts_tx_of_million_coins_and_pos_wallet_receives()
         {
-            TestHelper.WaitLoop(() => this.nodes[PosStaker].CreateRPCClient().GetRawMempool().Length > 0);
-            TestHelper.WaitLoop(() => this.nodes[PosStaker].FullNode.WalletManager().GetSpendableTransactionsInWallet(PosWallet).Any());
-
-            this.sharedSteps.WaitForNodeToSync(this.nodes[PosStaker]);
-
-            var received = this.nodes[PosStaker].FullNode.WalletManager().GetSpendableTransactionsInWallet(PosWallet);
-            received.Sum(s => s.Transaction.Amount).Should().Be(Money.COIN * 1000000);
+            this.proofOfStakeSteps.PowWalletBroadcastsTransactionToPosWallet();
         }
 
         private void pos_node_mines_ten_blocks_more_ensuring_they_can_be_staked()
         {
-            this.nodes[PosStaker].GenerateStratisWithMiner(Convert.ToInt32(this.nodes[PosStaker].FullNode.Network.Consensus.Option<PosConsensusOptions>().CoinbaseMaturity));
+            this.proofOfStakeSteps.PosNodeMinesToEnsureStaking();
         }
 
         private void pos_node_starts_staking()
         {
-            var minter = this.nodes[PosStaker].FullNode.NodeService<IPosMinting>();
-            minter.Stake(new WalletSecret() { WalletName = PosWallet, WalletPassword = PosWalletPassword });
+            this.proofOfStakeSteps.PosNodeMinesToEnsureStaking();
```

Comment:
this is the same method call as the step above mining 10 blocks to allow staking, was that a typo ?

Reply:
Good spot! Changed.